Sources/SwiftParser/Types.swift

```diff
@@ -1069,6 +1069,20 @@ extension Parser {
             arena: self.arena
           )
         )
+      } else if let colon = self.consume(if: .colon) {
```

Comment:
I’m a little worried that we enter into this path and give a weird diagnostic if somebody tries to do a Python-style function declaration like the following. It would probably be better to only recover to a dictionary type if we have colon followed by a type. And yes, I know that this doesn’t match the behavior of the old parser.

```swift
func foo() -> Int:
  // some Code
```

Reply:
Something like this. 
Also added a test case for the scenario above.